internal/core/src/common/GeometryCache.h

```diff
@@ -33,62 +33,66 @@ struct SegmentFieldHash {
     }
 };
 
-// Simple WKB-based Geometry cache for avoiding repeated WKB->Geometry conversions
+// Vector-based Geometry cache that maintains original field data order
 class SimpleGeometryCache {
  public:
-    // Get or create Geometry from WKB data
-    std::shared_ptr<const Geometry>
-    GetOrCreate(const std::string_view& wkb_data) {
-        // Use WKB data content as key (could be optimized with hash later)
-        std::string key(wkb_data);
-
-        // Try read-only access first (most common case)
-        {
-            std::shared_lock<std::shared_mutex> lock(mutex_);
-            auto it = cache_.find(key);
-            if (it != cache_.end()) {
-                return it->second;
-            }
-        }
-
-        // Cache miss: create new Geometry with write lock
+    // Append WKB data during field loading
+    void
+    AppendData(const char* wkb_data, size_t size) {
```

Comment:
can we do copy on write so read don't need any lock?

Reply:
or  we can at least do geometry_cache.rlock(), so we don't need to get read lock when process every data

Reply:
support  geometry_cache.RLockCache and GetByoffsetUnsafe.